cores/nRF5/Uart.cpp

```diff
@@ -64,12 +64,22 @@ void Uart::begin(unsigned long baudrate)
 
 void Uart::begin(unsigned long baudrate, uint16_t /*config*/)
 {
+#ifdef NRF52805_XXAA
```

Comment:
I was working on nrf52810 and I started with adding NRF528**10**_XXAA to this ifdef (and that works) but realized that complicated things when you need to support multiple chips ...

My [solution](https://github.com/sureshsitaula/arduino-nRF5/commit/0bd1c2aa31e5a8cb0c5dc3e07f07f9b9a1c186f8) was to edit the compat header (cores/nRF5/SDK/components/device/nrf52_to_nrf52810.h) instead and I didn't need to modify this file.
(and this matches the existing implementation for some other chips, see [nrf52840](https://github.com/sandeepmistry/arduino-nRF5/blob/master/cores/nRF5/SDK/components/device/nrf52_to_nrf52840.h) for example).

Should we update the compat header instead of this file?

Reply:
Should we be touching code in the SDK? Does that introduce potential upgrade complexity?

Reply:
Good question for @sandeepmistry :)
It would be nice to have this consistency between compat .h files, like [nrf52_to_nrf52840.h](https://github.com/sandeepmistry/arduino-nRF5/blob/master/cores/nRF5/SDK/components/device/nrf52_to_nrf52840.h) referenced above.

Reply:
> Should we be touching code in the SDK? Does that introduce potential upgrade complexity?

It would be better to avoid doing this, as it would be one more thing to manage when updating the SDK.

That said if it's the best approach wrapping changes in `#ifdef ARDUINO` would be best.

Reply:
Am I good to leave this as-is for now? There is already series and in some special cases chip specific logic in the .cpp library files. I would prefer to leave this condition where it is until we decide on the best way/place to extend SDK definitions without modifying the SDK.

However, I'll do whatever you feel is best. There is increased interested to get this merged in, so whatever I can do to help expedite.